src/ORM/DataObject.php

```diff
@@ -3175,14 +3175,26 @@ public static function reset()
     /**
      * Return the given element, searching by ID
      *
-     * @param string $callerClass The class of the object to be returned
-     * @param int $id The id of the element
-     * @param boolean $cache See {@link get_one()}
+     * @param string|int $callerClassOrID The class of the object to be returned (optional if called from extending class)
+     * @param int|bool $id The id of the element or bool to represent cache if ID is passed as fist argument
+     * @param bool $cache See {@link get_one()}
      *
-     * @return DataObject The element
+     * @return static The element
      */
-    public static function get_by_id($callerClass, $id, $cache = true)
+    public static function get_by_id($callerClassOrID, $id = null, $cache = true)
     {
+        if (is_numeric($callerClassOrID)) {
+            // Alternate signature to allow calling as MyDataObject::get_by_id($id)
+            $cache = is_null($id) ? true : (bool)$id;
+            $id = $callerClassOrID;
+            $callerClass = get_called_class();
+            if (__CLASS__ === $callerClassOrID) {
```

Comment:
This error is impossible if `is_numeric($callerClassOrID)`. Did you mean `if (__CLASS___ === $callerClass)`?

Reply:
This error can be outside of the arg processing loop anyway.